cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/io/cdap/cdap/etl/spec/PipelineSpecGenerator.java

```diff
@@ -348,6 +348,13 @@ protected ConfiguredStage configureStage(ETLStage stage, ValidatedPipeline valid
         .withStacktrace(e.getStackTrace());
       // Log the NullPointerException for debugging:
       LOG.error(String.format("Null error occurred while configuring the stage %s.", stageName), e);
+    } catch (ArrayIndexOutOfBoundsException e) {
+      // handle the case where plugin throws index out of bounds exception,
+      // this is to avoid having a number like '2', '8' etc as error message
+      collector.addFailure(String.format("Index out of bounds error occurred while configuring the stage %s.",
+                                         stageName), null).withStacktrace(e.getStackTrace());
+      // Log the ArrayIndexOutOfBoundsException for debugging:
+      LOG.error(String.format("ArrayIndexOutOfBounds error occurred while configuring the stage %s.", stageName), e);
```

Comment:
Why log? the error is already responded back to the client and it is a client side error?

Reply:
I was referring to the way Null Pointer Exceptions are caught 1 line above this. Let me know if it needs to be changed.

Reply:
I think we shouldn't log this.

Reply:
Removed